### PR TITLE
feat: add course_type to search/all response json

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

The `course_type` field has been added to `Course` objects distinguish new 2U additions to our catalog, like Executive Education and Bootcamp content. Downstream consumers of Discovery - like Enterprise Catalog Service - require `search/all` responses to contain the new field for indexing/filtering.

## References

- Reverts openedx/course-discovery#3459
- Previous attempt openedx/course-discovery#3423
- Previous attempt openedx/course-discovery/pull/3420
- [ENT-5823](https://2u-internal.atlassian.net/browse/ENT-5823)
- https://github.com/openedx/course-discovery/pull/3351
- https://github.com/openedx/enterprise-catalog/pull/448

### CODE REVIEW

**Comment by DawoudSheraz** [comment]
After checking out the branch on the local and running the following in the Discovery shell, I was able to see the course_type field in http://localhost:18381/api/v1/search/all/.

- ./manage.py search_index --disable-change-limit --create
- ./manage.py search_index --disable-change-limit --populate

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py [hunk 1]
```diff
@@ -334,10 +334,12 @@ def test_ingest_handles_draft(self, official_exists, draft_exists, partner_uses_
         all_courses = set()
         all_runs = set()
         audit_run_type = CourseRunType.objects.get(slug=CourseRunType.AUDIT)
+        course_type = CourseType.objects.get(slug=CourseType.AUDIT)
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/data_loaders/tests/test_api.py:337]: why is this needed/how is it related to the mani changes of this PR?

**Review by DawoudSheraz — APPROVED** [review]

File: course_discovery/apps/course_metadata/data_loaders/tests/test_api.py (hunk 1 above)
Comment (johnnagro) [reply course_discovery/apps/course_metadata/data_loaders/tests/test_api.py:337]: the new course_type field is required by these tests after the other changes made
